nifi-mock/src/main/java/org/apache/nifi/util/TestRunner.java

```diff
@@ -61,6 +61,13 @@
      */
     ProcessContext getProcessContext();
 
+    /**
+     * Call the {@link org.apache.nifi.annotation.lifecycle.OnScheduled} /
+     * {@link org.apache.nifi.annotation.lifecycle.OnUnscheduled} methods, to simulate a processor
+     * that is enabled but never triggered.
+     */
+    void doNotRun();
```

Comment:
This is a very useful test method, but the method name is not very intuitive.  Instead of trying to encapsulate multiple methods, what do you think about breaking this up to two methods: `schedule()` and `unschedule()`?  That would correspond to the annotations.

Reply:
Taking another look at the options, an alternative solution might be to allow `0` as an acceptable number of `iterations` when calling the `run()` method.  Right now that value must be `1` or greater, but with an iteration number of `0`, it looks like the TestRunner would call `OnScheduled` and `OnUnscheduled`, which would exercise the behavior described.

Reply:
I would agree with you here @exceptionfactory. There already exists an `unSchedule` method on TestRunner, it appears. And an `onStopped`. Would recommend just added a `schedule` method - or allowing `0` as an acceptable number of iterations to the `run` method.

Reply:
Thanks.  The `run(0)` solution seems obvious in retrospect; I made that change.